spec/integration/helpers.rb

```diff
@@ -0,0 +1,22 @@
+def apply_manifest(manifest_text)
+  # Applies the given manifest in as safe a manner as possible.
+  # Note: This will try (and fail) to create files, etc.
+  Puppet[:code] = manifest_text
+  node = Puppet::Node.indirection.find(Facter.value(:fqdn))
+  catalog = Puppet::Resource::Catalog.indirection.find(
+    node.name, :use_node => node)
+  catalog = catalog.to_ral
+  catalog.finalize
+  # If we're a host_config (the default), puppet tries to store stuff.
+  catalog.instance_variable_set(:@host_config, false)
+  catalog.apply
+end
+
+def unconfine(provider_class, confine_values)
```

Comment:
A "confiner" is a thing that decides whether a particular resource provider is applicable. When we specify a command in the provider (using `has_command`, for example), puppet helpfully adds a confine entry for it so that the provider is only available on systems that have that command. While this is a good idea in general, we don't want that restriction when we run the tests because the test environment doesn't necessarily have glusterfs installed.

I'll add a comment containing a less verbose explanation. :-)